lib/jwt/algos/eddsa.rb

```diff
@@ -3,7 +3,7 @@ module Algos
     module Eddsa
       module_function
 
-      SUPPORTED = %w[ED25519].freeze
+      SUPPORTED = %w[Ed25519].freeze
```

Comment:
Does this change the value that is added to the `alg` header in the payload?

Reply:
Not 100% sure but I think that the alg header when using this algorithm + curve needs to be `EdDSA` as #334 suggests.

I suggest fixing this as a separate thing. Also if there are some other services/components that are assigning the alg header as `Ed25519` it would be nice to know the reasoning behind that.

Reply:
@anakinj if so, that should be a separate PR.

Reply:
Yes, in a separate PR and maybe it needs to go into the next major release.

Changing the generated `alg` is a breaking change and we should be a little careful, even if the current way would be totally wrong there could be some Ruby JWT-JWT implementations trusting that the alg is what it is.

Reply:
Are you ok with the change above?

Reply:
I'd rather not change the casing on the algorithm, but fix it properly in the next major release.

Reply:
Ok, reverted.